builders/ethereum/dev-env/thirdweb.md

```diff
@@ -110,184 +112,206 @@ thirdweb offers SDKs for a range of programming languages, such as React, React
 
-```javascript
-import { ThirdwebProvider } from '@thirdweb-dev/react';
-import { Moonbeam } from '@thirdweb-dev/chains';
+```typescript title="client.ts"
+import { createThirdwebClient } from "thirdweb";
+
+// Replace this with your client ID string
+// refer to https://portal.thirdweb.com/typescript/v5/client on how to get a client ID
+const clientId = import.meta.env.VITE_TEMPLATE_CLIENT_ID;
+
+export const client = createThirdwebClient({
+  clientId: clientId,
+});
 
-const App = () => {
-  return (
-    <ThirdwebProvider activeChain={Moonbeam}>
-      <YourApp />
-    </ThirdwebProvider>
-  );
-};
+```
+
+### Configure Chain {: #configure-chain }
+thirdweb offers a small number of chains from `@thirdweb/chains` and does not include Moonbeam networks in that list, so you'll need to specify the network details including chain ID and RPC URL. You can create a custom chain with [`defineChain`](https://portal.thirdweb.com/references/typescript/v5/defineChain){target=\_blank} as follows:
+
+```typescript title="App.tsx"
+import { defineChain } from "thirdweb";
+
+  const moonbase = defineChain({
+  id: BigInt(1287),
+  rpc: "https://moonbase-rpc.dwellir.com",
+})
 ```
 
 ### Get Contract {: #get-contract }
 
-To connect to your contract, use the SDK’s [`getContract`](https://portal.thirdweb.com/typescript/sdk.thirdwebsdk.getcontract){target=\_blank} method.
+To connect to your contract, use the SDK’s [`getContract`](https://portal.thirdweb.com/references/typescript/v5/getContract){target=\_blank} method.  As an example, let's fetch data from an [Incrementer contract on Moonbase Alpha](https://moonbase.moonscan.io/address/0xa72f549a1a12b9b49f30a7f3aeb1f4e96389c5d8){target=\_blank}.
 
-```javascript
-import { useContract } from '@thirdweb-dev/react';
+```typescript title="App.tsx"
+import { getContract } from "thirdweb";
+import { client } from "./client";
 
-function App() {
-  const { contract, isLoading, error } = useContract('INSERT_CONTRACT_ADDRESS');
-}
+const myContract = getContract({
+  client,             
+  chain: moonbase,       
+  address: INSERT_CONTRACT_ADDRESS,   
+  abi: INSERT_ABI               
+});
 ```
 
 ### Calling Contract Functions {: #calling-contract-functions }
 
-For extension based functions, use the built-in supported hooks. There are several hooks available for you to use, the following are a few examples:
+To call a contract in the latest version of the SDK, you can use [`prepareContractCall`](https://portal.thirdweb.com/typescript/v5/transactions/prepare){target=\_blank}.
 
-- Use the NFTs extension to access a list of NFTs owned by an address via the [`useOwnedNFTs` hook](https://portal.thirdweb.com/react/react.useownednfts){target=\_blank}:
+```typescript title="App.tsx"
+import { prepareContractCall } from "thirdweb";
 
-    ```javascript
-    import { useOwnedNFTs, useContract, useAddress } from '@thirdweb-dev/react';
+const tx = prepareContractCall({
+          contract,
```

Comment:
this is `myContract` in the previous section, so just need to either change this instance or that one